src/scenes/start-scene.js

```diff
@@ -142,12 +143,40 @@ export class StartScene extends Phaser.Scene {
             }
         });
 
-
+        this.randomizeButton = new TextButton(this, 50, 300, 'RANDOMIZE SEATING', {
+            onClick: () => {
+                // Make an array all the seats and randomly pick them off for each player
+                this.removeInactivePlayers() // FIXME normally for every player, there are ~2 additional inactive players. This manages to remove all of them
+                let emptySeats = []
+                this.seats.forEach(seat => {
+                    emptySeats.push(seat);
+                });
+                console.log('players: ', this.playersLabel.playerLabels) // FIXME used for debugging inactive player problem
+                this.playersLabel.playerLabels.forEach(label => {
+                    let randomIndex = Math.floor(Math.random()*emptySeats.length);
+                    let next_seat = emptySeats.splice(randomIndex, 1)[0]
+                    next_seat.add(label);
+                });
+            }
+        });
+        this.add.existing(this.randomizeButton);
+        this.randomizeButton.setVisible(false);
     }
 
 
     onPlayersUpdate(playersList) {
         console.log("Players update!");
         this.playersLabel.updateWithPlayers(playersList);
+        this.removeInactivePlayers() // FIXME normally for every player, there are ~2 additional inactive players. This manages to reduce it to ~1 additional inactive player per player
+    }
+
+    // FIXME should ultimately remove
+    removeInactivePlayers() {
+        var i;
+        for (i = 0; i < this.playersLabel.playerLabels.length; i++) {
```

Comment:
I think you should be able to use `filter` for this.
```
this.playersLabel.playerLabels = this.playersLabel.playerLabels.filter(label => label.active);
```

Reply:
Much cleaner, thanks! Done. I left removeInactivePlayers() as a separate function to make it more readable.